HW5(OOP_vs_FP)/HW5_Trupkina.js

```diff
@@ -0,0 +1,236 @@
+
+// Write two solutions for each problem.
+// _.1 in an iterative style (no higher-order functions)
+// _.2 applying higher-order functions
+
+// 1.
+const arrBirthday = [1998, 1985, 2001, 2006];
+
+//1.1
+function findAgeIn2021(arr) {
+    const arr2021 = [];
+    for (let i = 0; i < arr.length; i++) {
+        arr2021.push(2021 - arr[i]);
+    }
+    return arr2021;
+}
+console.log(findAgeIn2021(arrBirthday));
+//(4) [23, 36, 20, 15]
+
+//1.2 hof
+//create array fhat contain current age in 2021 with hof
+function findAgeIn2021_hof(arr, fn) {
+    const arr2021 = [];
+    for (let i = 0; i < arr.length; i++) {
+        arr2021.push(fn(arr[i]));//call special func for calculate age 
+    }
+    return arr2021;
```

Comment:
what if we remove iteration and put the work with higher-order functions into the code right away, without expecting that it will come from the outside?